pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +501,168 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
+								return 1;
+							} else if ($fieldName != null) {
+								result = (int) ($fieldName.length() - thatObj.$fieldName.length());
+							}
+							if(result != 0) {
+								return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.STRING ||
+						f.type() == Field.FieldType.ENUM ||
+						f.parent() == null /* Process a sub-message */) {
+					generatedCodeSoFar += (
+							"""
+                            if ($fieldName == null && thatObj.$fieldName != null) {
+                                return -1;
+                            }
+                            if ($fieldName != null &&  thatObj.$fieldName == null) {
+                                return 1;
+                            }
+                            if ($fieldName != null) {
+                                result = $fieldName.compareTo(thatObj.$fieldName);
+                            }
+                            if(result != 0) {
+                                return result;
+                            }
+                            """).replace("$fieldName", f.nameCamelFirstLower());
+				}  else {
+					throw new IllegalArgumentException("Unexpected field type for getting HashCode - " + f.type().toString());
+				}
+			}
+		}
+		return generatedCodeSoFar.indent(DEFAULT_INDENT * 2);
+	}
+
+	/**
+	 * Generates the compareTo code for a primitive wrapper field.
+	 * @param f The field for which to generate the compareTo code.
+	 * @return The generated code for compareTo method body
+	 */
+	private static String getPrimitiveWrapperCompareToGeneration(Field f) {
+		final String template =
+				"""
+                    if ($fieldName == null && thatObj.$fieldName != null) {
+                        return -1;
+                    } else if ($fieldName != null && thatObj.$fieldName == null) {
+                        return 1;
+                    } else if ($fieldName != null) {
+                        result = $compareStatement;
+                    }
+                    if(result != 0) {
+                        return result;
+                    }
+                    """;
+
+
+		final String compareStatement;
+		switch (f.messageType()) {
+			case "StringValue" ->
+					compareStatement =
+							"$fieldName.compareTo(thatObj.$fieldName)";
+			case "BoolValue" ->
+					compareStatement =
+							"java.lang.Boolean.compare($fieldName, thatObj.$fieldName)";
+			case "Int32Value", "UInt32Value" ->
+					compareStatement =
+							"java.lang.Integer.compare($fieldName, thatObj.$fieldName)";
+			case "Int64Value", "UInt64Value" ->
+					compareStatement =
+							"java.lang.Long.compare($fieldName, thatObj.$fieldName)";
+			case "FloatValue" ->
+					compareStatement =
+							"java.lang.Float.compare($fieldName, thatObj.$fieldName)";
+			case "DoubleValue" ->
+					compareStatement =
+							"java.lang.Double.compare($fieldName, thatObj.$fieldName)";
+			case "BytesValue" ->
```

Comment:
I believe this will fail to work correctly for many of the current `bytes` values in hedera-services messages (e.g. [ContractID](https://github.com/hashgraph/hedera-protobufs/blob/add-pbj-types-for-state/services/basic_types.proto#L155C13-L155C13)), which will need byte-wise comparison of the array content in order to be deterministic.
It's entirely reasonable to refuse to implement `Comparable` for any message with a `bytes` field (and refuse to compile that if `Comparable` is requested), at least initially, but comparing on length only seems very likely to create hard-to-debug ISS events.

Reply:
Got it, thank you. I will remove support for Bytes

Reply:
We may still need to support Bytes (ContractID comes to mind) eventually; but it can be a subsequent PR, and when we do we *must* implement byte-wise comparison in order to ensure a correct compareTo for messages with bytes fields.

Reply:
I think we will need the bytes for Account ID, otherwise we won't be able to use the feature in services.

Reply:
Right `AccountID` and `ContractID` both may contain `bytes`.

Reply:
okay, I will add support for bytes then.

Reply:
I believe this method still only compares the lengths for bytes at line 701 below.

Reply:
@anthony-swirldslabs should be fixed now